### PR TITLE
Fix ubuntu issue #175 and #176

### LINKED ISSUE #176: [Bug] [Ubuntu] secureboot testing fail due to a question about whether to replace serial port file
##### SUMMARY

```
2021-11-25 10:15:31,025 | Failed at Play [secureboot_enable_disable] *****************

2021-11-25 10:15:31,025 | TASK [Set VM power state to 'powered-on'] ******************
task path: /home/worker/workspace/Ansible_Autoinstall_Ubuntu_21.04/ansible-vsphere-gos-validation/common/vm_set_power_state.yml:34
fatal: [localhost]: FAILED! => {
    "answer": null,
    "changed": true,
    "instance": {
        "advanced_settings": {
         ...
        "guest_consolidation_needed": false,
        "guest_question": {
            "_vimtype": "vim.vm.QuestionInfo",
            "choice": {
                "_vimtype": "vim.option.ChoiceOption",
                "choiceInfo": [
                    {
                        "_vimtype": "vim.ElementDescription",
                        "key": "0",
                        "label": "button.serial.file.append",
                        "summary": "Append"
                    },
                    {
                        "_vimtype": "vim.ElementDescription",
                        "key": "1",
                        "label": "button.serial.file.overwrite",
                        "summary": "Replace"
                    },
                    {
                        "_vimtype": "vim.ElementDescription",
                        "key": "2",
                        "label": "button.serial.file.cancel",
                        "summary": "Cancel"
                    }
                ],
                "defaultIndex": 0,
                "valueIsReadonly": null
            },
            "id": "21233",
            "message": [
                {
                    "_vimtype": "vim.vm.Message",
                    "argument": [
                        "serial.log"
                    ],
                    "id": "msg.serial.file.open",
                    "text": "The serial port output file \"serial.log\" already exists. Do you want to replace it with new content or append new content to the end of the file?"
                }
            ],
            "text": "The serial port output file \"serial.log\" already exists. Do you want to replace it with new content or append new content to the end of the file?"
        },
```

##### ISSUE TYPE
- Bug Report

##### ANSIBLE VERSION

```paste below



```

##### CONFIGURATION

```paste below



```

### PR DESCRIPTION
**Pull request by keirazhang**
In Ubuntu 21.10, the network interface name is changed, not ens192 any more. So remove the network interface config from cloud-init user-data for autoinstall, and remove the serial port after OS installation to fix #176

### CODE REVIEW

File: common/vm_remove_serial_port.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Add a serial port to VM by using output file
+# Parameter:
+#   vm_serial_port_file_path: The serial port output file on datastore.
+
+- name: "Add a serial port using output file"
```
Comment (Tomorrow9) [inline common/vm_remove_serial_port.yml:8]: Task description and 'name' part should be changed for "Remove" operation.

File: linux/guest_customization/check_network_config.yml [hunk 2]
```diff
@@ -108,28 +108,40 @@
...
-
-        - name: "Set cloud-init GOSC network configuration files on Ubuntu/Debian OS"
-          set_fact:
-            src_network_file: "/etc/network/interfaces.d/50-cloud-init.cfg"
-          when: "gosc_workflow == 'cloud-init'"
-      when: >
-        (guest_os_ansible_distribution == "Ubuntu" and guest_os_with_gui is defined and guest_os_with_gui) or
-        (guest_os_ansible_distribution != "Ubuntu")
-  when: guest_os_family == "Debian"
+        src_network_file: "/etc/netplan/99-netcfg-vmware.yaml"
```
Comment (Tomorrow9) [inline linux/guest_customization/check_network_config.yml:115]: What about the netplan config file got from this task, is it not applicable here? And the file recorded here can cover all the versions?
```
# Get netplan config file for Ubuntu server    
- include_tasks: ../utils/get_netplan_config_file.yml
```

Comment (keirazhang) [reply linux/guest_customization/check_network_config.yml:115]: Take Ubuntu Server as an example,  get_netplan_config_file.yml returns 00-installer-config.yaml, which is the original network config file. But GOSC will add a new network config file: 99-netcfg-vmware.yaml for perl GOSC and 50-cloud-init.yaml for cloud-init GOSC. So, here we don't use get_netplan_config_file.yml

Comment (Tomorrow9) [reply linux/guest_customization/check_network_config.yml:115]: Ok, if it's the same on all the Ubuntu versions.

**Review by Tomorrow9 — APPROVED** [review]
LGTM, thanks.

**Review by Tomorrow9 — APPROVED** [review]

**Pull request merged**